server/test_main.py

```diff
@@ -1,41 +1,32 @@
-import json
-
-from json.decoder import JSONDecodeError
-from urllib import request
-from main import app  # importing app from main.py
-from fastapi.testclient import TestClient
+import pytest
 from main import app
+from httpx import AsyncClient
 
-client = TestClient(app)
-
+request_body = {
+    "species": [
+        {
+            "molecule": "CO",
+            "mole_fraction": 0.2
+        }
+    ],
+    "mode": "absorbance",
+    "database": "hitran",
+    "tgas": 300,
+    "min_wavenumber_range": 1900,
+    "max_wavenumber_range": 2300,
+    "pressure": 1.01325,
+    "path_length": 11,
+    "use_simulate_slit": True,
+    "simulate_slit": 5,
+    "wavelength_units": "1/u.cm",
+    "path_length_units": "u.km",
+    "pressure_units": "cds.atm"
+}
 
 # testing for calculate_spectrum endpoint
-def test_calculate_spectrum():
-    try:
-        with open('./mocks/response.json', 'r') as f:
-            client_response = json.load(f)
-            responses = json.dumps(client_response)
-        response = client.post("/calculate-spectrum/", headers={'Content-Type': 'application/json'}, json={
-            "species": [
-                {
-                    "molecule": "CO",
-                    "mole_fraction": 0.2
-                }
-            ],
-            "mode": "absorbance",
-            "database": "hitran",
-            "tgas": 300,
-            "min_wavenumber_range": 1900,
-            "max_wavenumber_range": 2300,
-            "pressure": 1.01325,
-            "path_length": 11,
-            "use_simulate_slit": "true",
-            "simulate_slit": 5,
-            "wavelength_units": "1/u.cm",
-            "path_length_units": "u.km",
-            "pressure_units": "cds.atm"
-        })
-        # assert response.status_code == 200
-        assert response.json() == responses
-    except JSONDecodeError as e:
-        print("Error calculating")
+@pytest.mark.anyio
+async def test_calculate_spectrum():
+    async with AsyncClient(app=app, base_url="https://api.radis.app") as ac:
+        response = await ac.post("/calculate-spectrum", json=request_body, headers={"Content-Type": "application/json"})
+        assert response.status_code == 200 
+#TODO: add a efficient way to cmapare mock response with actual response
```

Comment:
Why can't the responses be compared?

Reply:
I feel comparison takes a lot of time in ci build and also it is kind of a big JSON that we are mocking means if there is any issue (indenting or value ) we get the  build failed, it is kind of hard to debug, although I think to compute the JSON value in tests itself and compare those values with the API response.

Reply:
feel free to drop your thoughts !! 😀